core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -879,30 +879,40 @@ class MapUnit {
         return damageFactor
     }
 
-    private fun getTerrainDamage() {
-        // hard coded mountain damage for now
-        if (getTile().baseTerrain == Constants.mountain) {
-            val tileDamage = 50
-            health -= tileDamage
+    private fun doTerrainDamage() {
+        val tileDamage = getDamageFromTerrain()
+        health -= tileDamage
 
-            if (health <= 0) {
-                civInfo.addNotification(
-                    "Our [$name] took [$tileDamage] tile damage and was destroyed",
-                    currentTile.position,
-                    name,
-                    NotificationIcon.Death
-                )
-                destroy()
-            } else civInfo.addNotification(
-                "Our [$name] took [$tileDamage] tile damage",
+        if (health <= 0) {
+            civInfo.addNotification(
+                "Our [$name] took [$tileDamage] tile damage and was destroyed",
                 currentTile.position,
-                name
+                name,
+                NotificationIcon.Death
             )
+            destroy()
+        } else if (tileDamage > 0) civInfo.addNotification(
+            "Our [$name] took [$tileDamage] tile damage",
+            currentTile.position,
+            name
+        )
+    }
+
+    fun getDamageFromTerrain(terrainName: String = currentTile.baseTerrain): Int {
+        var totalDamage = 0
+        if (terrainName == Constants.mountain)
+            totalDamage += 50   // The norm
```

Comment:
If we're renovating all this fun stuff, this should be a unique as well.

Reply:
The thing is, 50 damage from mountains is actually not unique to Carthage at all but rather shared for all civs, relevant since any civ can enter a mountain with a helicopter. Hence the half-implemented unique.
If we want to get rid of the hardcoding, we should probably add a `damagePerTurn` field to Terrains. The implementation of the unique `"Units ending their turn on a [] take [] damage"` would then be to overwrite this amount of damage for that civ. Meaning Carthages unique in this case will still be effectively meaningless since it means it will take 50 damage instead of 50 from mountains. So we can still set the `nonStandardTerrainDamage` to false for Carthage on base ruleset and save a bit of time in pathfinding.
The other alternative to get rid of hardcoded terrain damage would be to give _every_ civ a `"Units ending their turn on a [Mountain] take [50] damage"`, but that seems very inelegant.